### PR TITLE
Upgrade spacy v3

### LINKED ISSUE #167: Upgrade to spacy nightly?
Spacy 3.0 has native support to transformers 3.0.0. There seems that there are a couple of issues related do spacy not talking 
to the correct versions to transformers.

Should we update everything to 3.0.0?

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

Fixes #167 

Checklist
---

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

File: tests/test_bert_classifier.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # encoding: utf-8
 import numpy as np
 
-from wellcomeml.ml import BertClassifier
+from wellcomeml.ml.bert_classifier import BertClassifier
```
Comment (nsorros) [inline tests/test_bert_classifier.py:4]: remove this, i did during testin

File: setup.py [hunk 2]
```diff
@@ -61,9 +61,9 @@
         'deep-learning': [
             'tensorflow',
             'torch',
-            'transformers<2.9.0',
-            'spacy-transformers==0.6.1',
-            'dataclasses==0.6'  # spacy transformers needs this pinned
+            'transformers==3.0.0',
+#            'spacy-transformers==0.6.1',
```
Comment (nsorros) [inline setup.py:65]: remove this

**Comment by lizgzil** [comment]
fyi: The changes I made [here](https://github.com/wellcometrust/WellcomeML/pull/184/commits/7c77734c373f43a2985725a9f430157757e581db) are slowly fixing the entity linker, but not 100% yet. The current error is in the `.train` function, it happens when `optimizer = nlp.initialize(lambda: examples)` (line 75) is called:

```
OSError: [E050] Can't find model 'corpus/en_vectors'. It doesn't seem to be a Python package or a valid path to a data directory.
```

**Comment by lizgzil** [comment]
update about EL: On the recommendations of sophie I changed the initialisation method. Now my error (when running the test) is:
```
  File "spacy/tokens/doc.pyx", line 845, in sents
ValueError: [E030] Sentence boundaries unset. You can add the 'sentencizer' component to the pipeline with: `nlp.add_pipe('sentencizer')`. Alternatively, add the dependency parser or sentence recognizer, or set sentence boundaries by setting `doc[i].is_sent_start`.
```
nlp.pipe_names contains sentencizer, so I'm still trying to work this out

**Review by aCampello — CHANGES_REQUESTED** [review]
The tests pass, however after a conversation with @nsorros, before merging we want to check whether the code is backwards compatible, i.e. if model trained with v2.3 loaded into v3.0 works. It is quite important to know that, to inform what we'll do regarding this merge. If models are not backwards compatible, we might need to add some code during the migration period.

**Comment by nsorros** [comment]
> The tests pass, however after a conversation with @nsorros, before merging we want to check whether the code is backwards compatible, i.e. if model trained with v2.3 loaded into v3.0 works. It is quite important to know that, to inform what we'll do regarding this merge. If models are not backwards compatible, we might need to add some code during the migration period.

In regards to the trained SciBERT model it seems to fail
```
>>> model.load('models/scibert-2020.05.5')
/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py:646: UserWarning: [W094] Model 'en_model' (0.0.0) specifies an under-constrained spaCy version requirement: >=2.2.1. This can lead to compatibility problems with older versions, or as new spaCy versions are released, because the model may say it's compatible when it's not. Consider changing the "spacy_version" in your meta.json to a version range, with a lower and upper pin. For example: >=3.0.0rc2,<3.1.0
  warnings.warn(warn_msg)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/code/WellcomeML/wellcomeml/ml/bert_classifier.py", line 85, in load
    self.nlp = spacy.load(model_dir)
  File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/__init__.py", line 47, in load
    return util.load_model(name, disable=disable, exclude=exclude, config=config)
  File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 275, in load_model
    return load_model_from_path(Path(name), **kwargs)
  File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 339, in load_model_from_path
    config = load_config(config_path, overrides=dict_to_dot(config))
  File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 476, in load_config
    raise IOError(Errors.E053.format(path=config_path, name="config.cfg"))
OSError: [E053] Could not read config.cfg from models/scibert-2020.05.5/config.cfg
```

**Comment by nsorros** [comment]
> > The tests pass, however after a conversation with @nsorros, before merging we want to check whether the code is backwards compatible, i.e. if model trained with v2.3 loaded into v3.0 works. It is quite important to know that, to inform what we'll do regarding this merge. If models are not backwards compatible, we might need to add some code during the migration period.
> 
> In regards to the trained SciBERT model it seems to fail
> 
> ```
> >>> model.load('models/scibert-2020.05.5')
> /data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py:646: UserWarning: [W094] Model 'en_model' (0.0.0) specifies an under-constrained spaCy version requirement: >=2.2.1. This can lead to compatibility problems with older versions, or as new spaCy versions are released, because the model may say it's compatible when it's not. Consider changing the "spacy_version" in your meta.json to a version range, with a lower and upper pin. For example: >=3.0.0rc2,<3.1.0
>   warnings.warn(warn_msg)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/data/code/WellcomeML/wellcomeml/ml/bert_classifier.py", line 85, in load
>     self.nlp = spacy.load(model_dir)
>   File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/__init__.py", line 47, in load
>     return util.load_model(name, disable=disable, exclude=exclude, config=config)
>   File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 275, in load_model
>     return load_model_from_path(Path(name), **kwargs)
>   File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 339, in load_model_from_path
>     config = load_config(config_path, overrides=dict_to_dot(config))
>   File "/data/code/WellcomeML/build/virtualenv/lib/python3.6/site-packages/spacy/util.py", line 476, in load_config
>     raise IOError(Errors.E053.format(path=config_path, name="config.cfg"))
> OSError: [E053] Could not read config.cfg from models/scibert-2020.05.5/config.cfg
> ```

BertClassifier is rewritten so this is resolved. Essentially the strategy is to retrain the model.

**Review by aCampello — APPROVED** [review]

**Pull request merged**